tests/test_identify.py

```diff
@@ -1,26 +1,160 @@
+            while np.random.rand() < p:
+                x += str(np.random.choice(alphabet))
+            if f(x):
+                positives.add(x)
+            else:
+                negatives.add(x)
+        return list(positives), list(negatives)
 
-    accepting = [[0], [0, 'z', 0, 0], ['z', 'z']]
-    rejecting = [[0, 'z', 'z'], ['z']]
+    def generate_problem(self, n, input_sz, bypass = False):
+        def g(x):
+            ones = [a for a in x if a == '1']
+            return 1 if len(ones) % n == 0 else 0
+        accepting, rejecting = self.rand_test_set([0,1], g, input_sz)
+        if bypass:
+            return IdProblem(accepting, rejecting, 0)
+        problem = IdProblem(accepting, rejecting, n)
+        return problem
+
+    def solve_and_check(self, problem: IdProblem, symm_mode: SymmBreak = SymmBreak.CLIQUE):
+        accepting = problem.accepting
+        rejecting = problem.rejecting
+        
+        my_dfa = find_dfa(accepting=accepting, rejecting=rejecting, symm_mode = symm_mode)
+
+        # check that identified DFA fits observations
+        for x in accepting:
+            assert my_dfa.label(x)
+
+        for x in rejecting:
+            assert not my_dfa.label(x)
+
+        # check that returned DFA is minimal (check bypassed if problem.size = 0)
+        assert (problem.size == 0) or len(my_dfa.states()) == problem.size
+        return my_dfa
+
+    def test_identify(self):
+
+        symm = SymmBreak.BFS
+
+        self.solve_and_check(self.problem1, symm_mode = symm)
+
+        my_dfa = self.solve_and_check(self.problem2, symm_mode = symm)
+        write_dot(my_dfa, os.path.join(self.outdir,"test1.dot"))
+
+        my_dfa = self.solve_and_check(self.problem3, symm_mode = symm)
+        write_dot(my_dfa, os.path.join(self.outdir,"test2.dot"))
+
+        my_dfa = self.solve_and_check(self.problem5, symm_mode = symm)
+        write_dot(my_dfa, os.path.join(self.outdir,"test3.dot"))
     
-    my_dfa = find_dfa(accepting=accepting, rejecting=rejecting)
+    def test_identify_2(self):
+
+        input_sz = 200
+        
+        symm = SymmBreak.NONE
+        start = 0
+        end = 5
+        step = 1
+        problems = {}
+
+        for n in range(start, end, step):
+            problems[n] = self.generate_problem(8, input_sz, bypass=True)
+            # problems[n] = self.generate_problem(n, input_sz, bypass=True)
+
+        print("Running tests for no symmetry breaking...")
+        for n in range(start, end, step):
+
+            t0 = time.time()
+            dfa1 = self.solve_and_check(problems[n], symm_mode=symm)
+            elapsed = time.time() - t0
+            
+            print("solved n = {} with no symmetry-breaking in {:.4g}s".format(n,elapsed))
+            write_dot(dfa1, os.path.join(self.outdir,"longtest_n-{}_symm-{}.dot".format(n,symm)))
+        
+        symm = SymmBreak.CLIQUE
+        print("Running tests for clique-based symmetry breaking...")
+        for n in range(start, end, step):
```

Comment:
Can you break out the benchmarking into a different file? If we're checking for performance regressions, we can also set a deadline in hypothesis.